src/OnboardingSPA/pages/Steps/GetStarted/SiteTypeSetup/PrimarySite/index.js

```diff
@@ -37,130 +37,138 @@ const StepPrimarySetup = () => {
 		setIsDrawerSuppressed( true );
 		setDrawerActiveView( VIEW_NAV_GET_STARTED );
 		setIsHeaderNavigationEnabled( true );
+		getSiteClassificationsData();
 	}, [] );
 
-	const [ clickedIndex, changeCategory ] = useState( -1 );
-	const [ inputCategVal, changeInputCateg ] = useState( '' );
+	const [ siteClassData, setSiteClassData ] = useState();
+	const [ primaryCategory, changePrimaryCategory ] = useState( '' );
+	const [ inputFieldValue, setInputFieldValue ] = useState( '' );
 
-	const categoriesArray = content.categories;
-	const selectedPrimaryCategoryInStore = currentData?.data?.siteType?.primary;
+	/**
+	 * Function which fetches the Site Classifications
+	 *
+	 */
+	const getSiteClassificationsData = async () => {
+		let siteClassificationsData = await getSiteClassifications();
+		if ( siteClassificationsData?.body.length === 0 ) {
+			siteClassificationsData = fallbackDataTypes;
+		}
 
-	/**This condition fills the data in input box if the saved category isn't a subcategory from the content*/
-	if ( selectedPrimaryCategoryInStore && ! inputCategVal ) {
-		const found = categoriesArray.find(
-			( e ) => e.name === selectedPrimaryCategoryInStore
-		);
-		if ( ! found && selectedPrimaryCategoryInStore !== 'primaryCategory' )
-			changeInputCateg( selectedPrimaryCategoryInStore );
-	}
+		setSiteClassData( siteClassificationsData?.body );
+		changePrimaryCategory( currentData?.data?.siteType?.primary ?? '' );
+
+		if ( currentData?.data?.siteType?.labelPri === 'custom' ) {
+			changePrimaryCategory( '' );
+			setInputFieldValue( currentData?.data?.siteType?.primary );
+		}
+	};
```

Comment:
```suggestion
		setSiteClassData( siteClassificationsData?.body );
		if ( currentData?.data?.siteType?.labelPri === 'custom' ) {
			changePrimaryCategory( '' );
			setInputFieldValue( currentData?.data?.siteType?.primary );
			return;
		}
		changePrimaryCategory( currentData?.data?.siteType?.primary ?? '' );
	};
```
To prevent 2 state updates, but I'm sure this will change as part of consolidating `inputField` into `primaryCategory`, just adding this comment so that you keep this in mind when writing the new logic.

Reply:
Consolidating it might not work as an input field takes a `value` attribute that is expected to be string and the input box should always be rendered on the screen and will start showing the chip label, and doing a check for if custom should be done outside the value field.

Reply:
Still 2 state updates as the **custom state** object ( which tells if the custom text is being used ) and the **primary state** update happens